gnovm/pkg/gnolang/realm.go

```diff
@@ -453,10 +456,18 @@ func (rlm *Realm) incRefCreatedDescendants(store Store, oo Object) {
 				// NOTE: may already be marked for first gen
 				// newCreated or updated.
 				child.SetOwner(oo)
-				rlm.incRefCreatedDescendants(store, child)
+
+				// As a child of packageValue, packageBlock is
+				// visited once. As a child of fileBlock, it's
+				// visited twice. On the second visit, it's already
+				// Real, while it should be NewReal too, to avoid
+				// redundant dirty marking. So make SetNewReal happen
+				// first before recurse.
 				child.SetIsNewReal(true)
+				rlm.incRefCreatedDescendants(store, child)
```

Comment:
is this the only logical fix in this PR? 
I'm not sure why ModTime went from 0 to positive in the test files.
Doesn't it mean that something in this PR is marking things dirty MORE than before?

Reply:
see in the other comment.